### PR TITLE
improve Enumerate's docs, yields zero-based index

### PR DESCRIPTION
**Pull request by liigo**

### CODE REVIEW

File: library/core/src/iter/adapters/enumerate.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use crate::iter::adapters::{
 use crate::iter::{FusedIterator, InPlaceIterable, TrustedLen};
 use crate::ops::Try;
 
-/// An iterator that yields the current count and the element during iteration.
+/// An iterator that yields the zero-based index and the element during iteration.
```
Comment (JohnTitor) [inline library/core/src/iter/adapters/enumerate.rs:7]: I don't think this improves much, but this just rephrases the current one. Looking at the other docs, we don't usually mention the index is zero-based, see an Iterator itself or `enumerate` method's docs.
Also, the method docs' example shows that the count is zero-based and that's enough to explain, I think.

**Comment by anden3** [comment]
Hello @liigo! I'm pinging you as part of the triage procedure as you've received a review for this PR :)

**Comment by JohnCSimon** [comment]
@liigo 

Ping from triage: I'm closing this due to inactivity as it hasn't been interaction from the author in a few months. Please reopen when you are ready to continue with this. 
Note: if you are going to continue please open the PR **BEFORE** you push to it, else you won't be able to reopen - this is a quirk of github. 
Thanks for your contribution.

@rustbot label: +S-inactive

**Comment by liigo** [comment]
@JohnCSimon What are you expect me to interact? Since noone propose changing this PR, you (or other reviewers) are free to merge or refuse it. (I've read JohnTitor' review as refuse.)

**Comment by JohnCSimon** [comment]
@liigo I'm not a reviewer, triage is just to clean up deleted or abandoned PRs. - you can re-open the PR if you like.

**Pull request closed without merging**